lib/checkPullRequestJob.js

```diff
@@ -32,6 +32,101 @@ const JOB_REGEX = new RegExp(
   ].join('')
 );
 
+const CRON_REGISTRY_FILENAME = ['cron.yaml', 'main_cron.py', 'cron.py'];
+const CRON_JOB_REGEX = new RegExp(
+  [
+    '(?<addition>\\+)(?<classDefinition>class\\s)',
+    '(?<name>[a-zA-Z]{2,256})(?<suffix>Handler)(?<funDef>\\()'
+  ].join('')
+);
+
+/**
+ * @param {import('probot').Octokit.PullsListFilesResponseItem} file
+ */
+const getNewCronJobsFromFile = (file) => {
+  return getNewItemsFromFileByRegex(CRON_JOB_REGEX, file);
+};
+
+/**
+ * @param {import('probot').Octokit.PullsListFilesResponseItem} file
+ */
+const addsNewCronJob = (file) => {
+  const newJobs = getNewCronJobsFromFile(file);
+  return newJobs.length > 0;
+};
+
+const compare = (CRON_REGISTRY_FILENAME, changedFiles) => {
+  CRON_REGISTRY_FILENAME.forEach((e1) => changedFiles.forEach((e2) => {
+    if(e1 === e2) {
+      return true;
+    }
+  }));
+}
+
+const getCronCommentBody = (changedFiles, prAuthor) => {
+  const newLineFeed = '<br>';
+  const cronJobsForm = 'cron jobs form'.link('');// some link
+  const wikiLinkText = (
+    'this guide'.link(
+      'https://github.com/oppia/oppia/wiki/Running-jobs-in-production' +
+      '#submitting-a-pr-with-a-new-job'));
+
+  // This function will never be called when there are no job files.
+  if (compare(CRON_REGISTRY_FILENAME, changedFiles)) {
```

Comment:
Why do we need this anyway? We already check for the changes to files on line 108 right?

Reply:
@vojtechjelinek I think `changedFiles` should contain all the files that are changed in a particular PR? So if  suppose the changed files doesn't contain any of the cron files then the bot shouldn't get trigger .But if there are some files from the 
`CRON_REGISTRY_FILENAME` then it would let us know that any of the cron files have changed.Does that make sense?

Reply:
Yep, but on line 108 we use `newJobFiles` which are `changedFiles` filtered and should only contain the modified job files, right?